Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,309 @@
+                                break;
+                            case AxisType.SingleAxis:
+                                UpdateSingleAxisData(Interactions[i], controllerState);
+                                break;
+                            case AxisType.DualAxis:
+                                UpdateDualAxisData(Interactions[i], controllerState);
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateButtonDataPerfMarker = new ProfilerMarker("[MRTK] HPController.UpdateButtonData");
+
+        /// <summary>
+        /// Update an interaction bool data type from a bool input
+        /// </summary>
+        /// <remarks>
+        /// Raises an Input System "Input Down" event when the key is down, and raises an "Input Up" when it is released (e.g. a Button)
+        /// </remarks>
+        internal virtual void UpdateButtonData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateButtonDataPerfMarker.Auto())
+            {
+                // Handedness must be left or right in order to differentiate between buttons for the left and right hand.
+                MixedReality.Input.Handedness controllerHandedness = controllerState.MotionController.Handedness;
+
+
+                Debug.Assert(controllerHandedness != MixedReality.Input.Handedness.Unknown);
+                Debug.Assert(interactionMapping.AxisType == AxisType.Digital);
+
+                if (interactionMapping.InputType == DeviceInputType.TriggerTouch)
+                {
+                    var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                    interactionMapping.BoolData = !Mathf.Approximately(triggerData, 0.0f);
+                }
+                else if (interactionMapping.InputType == DeviceInputType.GripTouch)
+                {
+                    var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                    interactionMapping.BoolData = !Mathf.Approximately(gripData, 0.0f);
+                }
+                else
+                {
+                    ControllerInput button;
+
+                    // Update the interaction data source
+                    switch (interactionMapping.InputType)
+                    {
+                        case DeviceInputType.Select:
+                        case DeviceInputType.TriggerNearTouch:
+                        case DeviceInputType.TriggerPress:
+                            button = ControllerInput.Trigger;
+                            break;
+                        case DeviceInputType.GripNearTouch:
+                        case DeviceInputType.GripPress:
+                            button = ControllerInput.Grasp;
```

Comment:
There are a few paths here that I'm not sure are represented in the mappings list. Maybe not super important to remove if future classes that inherit from this might use them though.

Reply:
The selection types here are simply to direct the configured interaction mapping to the specific function that gathers that data.  If a mapping isn't intended to be used, it isn't required.  
But for completeness, in the provider itself, in case the interaction mapping is configured to act, then we ensure the provider can get the data.  Without it, nothing would happen.